src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,124 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const [newMovieError, setNewMovieError] = useState({
+    titleError: false,
+    imgUrlError: false,
+    imdbUrlError: false,
+    imdbIdError: false,
+  });
+
+  const valueCheck = newMovie.title.trim() === ''
+  || newMovie.imgUrl.trim() === ''
+  || newMovie.imdbUrl.trim() === ''
+  || newMovie.imdbId.trim() === '';
+
+  const handleChange = (name: string, newValue: string) => {
+    setNewMovie((prevMovie) => ({
+      ...prevMovie,
+      [name]: newValue,
+    }));
+  };
+
+  const isAddDisabled = !newMovie.title
+  || !newMovie.imgUrl
+  || !newMovie.imdbUrl
+  || !newMovie.imdbId;
+
+  const handleAdd = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (isAddDisabled) {
+      setNewMovieError({
+        titleError: true,
+        imgUrlError: true,
+        imdbUrlError: true,
+        imdbIdError: true,
+      });
+
+      return;
+    }
+
+    onAdd(newMovie);
+    setCount(count + 1);
```

Comment:
```suggestion
    setCount(prev => prev + 1);
```
if you compute new value depending on old one you should use callback